feat: make robots happy

gomesalexandre opened the pull request:
## Description

(the Mountain View ones)

## Issue (if applicable)

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

closes N/A

## Risk

> High Risk PRs Require 2 approvals

<!-----------------------------------------------------------------------------
Outline the scope of your changes and the risk associated with them. You must use your discretion as an engineer to determine the potential impact of your changes.

WARNING: If your PR introduces a new on-chain transaction or modifies an existing one, please add the "High-Risk" label to this PR and ask for 2 reviewers to approve it before merging.

E.g. an upgrade to `hdwallet` or core state management would be considered higher risk, and might require a full regression test. UI or isolated view changes, or something behind a feature flag may have near zero risk. Small bug fixes might require testing isolated to the specific fix.
------------------------------------------------------------------------------>

None

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

<!-----------------------------------------------------------------------------
We treat every PR to be merged with the same scrutiny as if we were merging directly to production.

Your PR will not be merged if you do not complete the sections below for our engineering and operations teams to test.
------------------------------------------------------------------------------>


### Engineering

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

- Every single URL in sitemap.txt is happy

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

- None

## Screenshots (if applicable)


<!-- av pr metadata
This information is embedded by the av CLI when creating PRs to track the status of stacks when using Aviator. Please do not delete or edit this section of the PR.
```
{"parent":"feat_legacy_connect_wallet","parentHead":"f2630338e6f783dfe6fc00a0f702840226ed8ce9","parentPull":9318,"trunk":"develop"}
```
-->

gomesalexandre commented on public/sitemap.txt:
```diff
@@ -1,11 +1,9 @@
 https://app.shapeshift.com
-https://app.shapeshift.com/#/connect-wallet
-https://app.shapeshift.com/#/demo/fox
-https://app.shapeshift.com/#/demo/defi
-https://app.shapeshift.com/#/demo/defi/earn
-https://app.shapeshift.com/#/demo/trade
-https://app.shapeshift.com/#/demo/assets
-https://app.shapeshift.com/#/demo/assets/bip122:000000000019d6689c085ae165831e93/slip44:0
-https://app.shapeshift.com/#/demo/assets/eip155:1/slip44:60
-https://app.shapeshift.com/#/demo/assets/eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d
-https://app.shapeshift.com/#/demo/assets/eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3
```
- Removes demo prefix since demo wallet is now gone
- Remove /defi which is now gone
- Adds rFOX
- Adds buy-crypto
- Does *not* add lending/pools since those are non-actionable
- Does *not* add markets, not sure how crawlers crawl SPAs but just in case, no point spewing Portals

NeOMakinG commented on public/sitemap.txt:
```diff
@@ -1,11 +1,9 @@
+https://app.shapeshift.com/#/rFOX
+https://app.shapeshift.com/#/trade
+https://app.shapeshift.com/#/buy-crypto
+https://app.shapeshift.com/#/assets/bip122:000000000019d6689c085ae165831e93/slip44:0
+https://app.shapeshift.com/#/assets/eip155:1/slip44:60
+https://app.shapeshift.com/#/assets/eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d
+https://app.shapeshift.com/#/assets/eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3
```
q: I don't know if it's useful but we can trade and pairs here, so we could index more pages with different pairs (maybe it would rank us up in SEO?)

gomesalexandre replied:
The more routes we add, I believe the better! Can't be worse than currently.

<img width="1371" alt="image" src="https://github.com/user-attachments/assets/9ea1c48f-da89-43c5-a1c9-bce323178ae8" />


On a tangent re: SEO, I think we could also:
- find a way to make title head work, not sure if we can do this without SSR, but all pages being references
- ensure our crawling is sane and scales - would be nice to have a pass on that in a follow-up

If you want to tag team with hpayne and own this, feel free to create an issue and bring it to the table! 💪🏽

NeOMakinG replied:
Pretty sure we can without SSR, maybe it would require some page generation or something similar

Added to my todo, I'll think about it and create an issue depending on what I find!

NeOMakinG reviewed (changes requested):
Missing pages (Even if we are not loading content in those, we need to index them):

```
https://app.shapeshift.com/#/limit

// Not sure we want them thought in the current state, but adding a few texts and tweaking the description of the page would have a positive impact plus cross links from blogs and KOLs

https://app.shapeshift.com/#/wallet
https://app.shapeshift.com/#/wallet/accounts
https://app.shapeshift.com/#/wallet/earn
https://app.shapeshift.com/#/wallet/activity

// We want them as well as market and portal page, crawlers are now capable of executing JS meaning we could benefit from the content of our API providers as well as `hn` tags and descriptions
http://localhost:3000/#/markets/category/tradingVolume
http://localhost:3000/#/markets/category/marketCap
http://localhost:3000/#/markets/category/trending
http://localhost:3000/#/markets/category/topMovers
http://localhost:3000/#/markets/category/recentlyAdded
```

I think the market page and one click defi category should be added too (I did read your description/comment):

- We don't have any infinite loading scroll, it will just fetch assets as a regular user, are we afraid of having a few more users?

That's necessary if we want to improve our SEO

gomesalexandre commented:
> https://app.shapeshift.com/#/wallet
> https://app.shapeshift.com/#/wallet/accounts
> https://app.shapeshift.com/#/wallet/earn
> https://app.shapeshift.com/#/wallet/activity

I actually think the wallet subroutes do *not* hurt to to be had, nice one! Not like they do much without a wallet, but more crawling is probably always good for SEO? 19d11a4c69

> http://localhost:3000/#/markets/category/tradingVolume
> http://localhost:3000/#/markets/category/marketCap
> http://localhost:3000/#/markets/category/trending
> http://localhost:3000/#/markets/category/topMovers
> http://localhost:3000/#/markets/category/recentlyAdded

You know what, let's probably add them! Doubt robots crawl that often anyway, one user fetching the initial viewport (we're using deferred loading on visible) shouldn't really do any wrong. Added in 644391bc64


Finally, added limit and claim in afaaa9fca0 - the latter is not visible as a button if you don't have a wallet connected, but if we're going for completeness, might as well?

NeOMakinG reviewed (approved):
Looks good now, claim bit won't hurt as we don't have any redirect on this one so, we can keep it as it is!

NeOMakinG commented on public/sitemap.txt:
```diff
@@ -1,11 +1,22 @@
+https://app.shapeshift.com/#/wallet/accounts
+https://app.shapeshift.com/#/wallet/earn
+https://app.shapeshift.com/#/wallet/activity
+http://app.shapeshift.com/#/markets
+http://app.shapeshift.com/#/markets/category/tradingVolume
+http://app.shapeshift.com/#/markets/category/marketCap
+http://app.shapeshift.com/#/markets/category/trending
+http://app.shapeshift.com/#/markets/category/topMovers
+http://app.shapeshift.com/#/markets/category/recentlyAdded
```
Can we replace `http` to `https` please?

gomesalexandre commented:
> I think unreachable pages would do the contrary of a positive thing, claim is probably not a page we should add as it's more like an account page not accessible without wallet, and without value (no text, no title, no description...) it's an useless page in term of SEO

I think wallet would be the exact same? In effect, it's just an empty state, which is the same as claims. But I'll let you be the judge on this one 🙏🏽 

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/550272d6-0a97-4726-93f5-a3132bfd3760" />

twblack88 commented:
🤖 🥫

NeOMakinG commented:
> > I think unreachable pages would do the contrary of a positive thing, claim is probably not a page we should add as it's more like an account page not accessible without wallet, and without value (no text, no title, no description...) it's an useless page in term of SEO
> 
> I think wallet would be the exact same? In effect, it's just an empty state, which is the same as claims. But I'll let you be the judge on this one 🙏🏽
> 
> <img alt="image" width="1728" src="https://private-user-images.githubusercontent.com/17035424/432819257-550272d6-0a97-4726-93f5-a3132bfd3760.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2MTI4MjcsIm5iZiI6MTc0NDYxMjUyNywicGF0aCI6Ii8xNzAzNTQyNC80MzI4MTkyNTctNTUwMjcyZDYtMGE5Ny00NzI2LTkzZjUtYTMxMzJiZmQzNzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE0VDA2MzUyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjYTc2Y2M0ZjZkZGE5NzY3OWVmN2MyNGY5OGFiM2Y1YWExMzk2OTI0ZjY5OTQ5Njk2NTkyNDgxNTU0Y2Y3MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.18SPdfOjIYf7UHegQXb0jyZLumgDTnUCXXU3VedZpZ8">

It doesn't count so much, it won't block anything but won't necessary do something given the current state of those page so, we can add them!
